gateway-shell/src/main/java/org/apache/knox/gateway/shell/table/KnoxShellTableFilter.java

```diff
@@ -32,7 +32,15 @@ public KnoxShellTableFilter table(KnoxShellTable table) {
   }
 
   public KnoxShellTableFilter name(String name) {
-    index = tableToFilter == null ? -1 : tableToFilter.headers.indexOf(name);
+    for (int i = 0; i < tableToFilter.headers.size(); i++) {
+      if (tableToFilter.headers.get(i).equalsIgnoreCase(name)) {
+        index = i;
```

Comment:
What is index used for outside of this method? Could there be an issue that index is being updated and changed to -1 on no matches before switched back to the right index?

Reply:
With the recent change - won't this case only ever be hit once? Since once you set `index` to something else it won't have the "magic" value of -1 to compare against again.

I think this method requires some more thought and some explanation of what the `index` variable is used for. Global state tends to be hard to reason about and if possible avoided.

Reply:
I see your original concern, my bad. `index` is a member variable of the filter class and only ever used in that one filter instance, not a global variable. As for -1, that is the returned index when no matches exist, which throws an exception.